koa-basic-auth/index.d.ts

```diff
@@ -0,0 +1,16 @@
+// Type definitions for koa-basic-auth 2.x
+// Project: https://github.com/koajs/basic-auth
+// Definitions by: Tobias Wolff <https://github.com/Tobias4872/>
+// Definitions: https://github.com/Tobias4872/DefinitelyTyped
+
+import * as Koa from "koa";
+
+declare function auth(opts?: {
```

Comment:
Well, I have been reviewing the code and the variable set to an empty object in case it is not passed. But it needs to contain the name and the pass property, otherwise, the code will trigger an assertion. Therefore, I'd rather suggest removing the optionality of the parameter. Would you agree?